src/server/routes/actions.ts

```diff
@@ -26,19 +33,86 @@ router.get<unknown, BalanceResponse>("/balance", (_, res) => {
     });
 });
 
+const missingParameterError = (
+  res: Response<DripErrorResponse>,
+  parameter: keyof BotDripRequestType | keyof FaucetRequestType,
+): void => {
+  res.status(400).send({ error: `Missing parameter: '${parameter}'` });
+};
+
+const addressMiddleware = (
+  req: Request<unknown, DripResponse, Partial<DripRequestType>>,
+  res: Response,
+  next: NextFunction,
+): void => {
+  if (!req.body.address) {
+    return missingParameterError(res, "address");
+  }
+  next();
+};
+
 const dripRequestHandler = new DripRequestHandler(actions, storage, recaptchaService);
 
+router.post<unknown, DripResponse, FaucetRequestType>("/faucet", addressMiddleware, async (req, res) => {
+  const { address, parachain_id, recaptcha } = req.body;
+  if (!recaptcha) {
+    return missingParameterError(res, "recaptcha");
+  }
+  try {
+    const dripResult = await dripRequestHandler.handleRequest({
+      external: true,
+      address,
+      parachain_id: parachain_id ?? "",
+      amount: config.Get("DRIP_AMOUNT"),
+      recaptcha,
+    });
+
+    if ((dripResult as DripErrorResponse).error) {
+      res.status(500).send(dripResult);
+    } else {
+      res.send(dripResult);
+    }
+  } catch (e) {
+    logger.error(e);
+    errorCounter.plusOne("other");
+    res.status(500).send({ error: "Operation failed." });
+  }
+});
+
+router.post<unknown, DripResponse, BotDripRequestType>("/drip-v2", addressMiddleware, async (req, res) => {
```

Comment:
few comments:
1. we're exposing now the public URL which you can use directly to call API endpoint to drip amount to any address and substitute any random sender (as it's not controlled by matrix now) and no captcha.. isn't it potential way for breach? )) does it make sense to check if the current server is indeed `config.Get("EXTERNAL_ACCESS") === true` ? because otherwise this should go with captcha token

2. as I see we don't validate parachain_id anywhere, so might worth validate user input?

Reply:
Oh god! You are right, I didn't thought about that.

I'm gonna add some kind of check to it to ensure that it doesn't go through.

Quoting @mutantcornholio: 
![image](https://user-images.githubusercontent.com/8524599/220432194-fbecf1c8-c54c-469a-8f49-f28ce56e189c.png)